### DISCUSSION TITLE
Trying to Log a message in Console when VL53L0X#Distance<=15

### DISCUSSION THREAD

**Discussion by jackpotcityco** [arendst/Tasmota#19723] [Q&A] [answered]
Hello,

I try to create my first Rule1 that look like this in the Console in the Web Interface.

`Rule1 ON VL53L0X#Distance<=15 DO Log "Distance is less than or equal to 15" ENDON`

What I try to do is, that when the Distance measures <=15 I want to log the message: "Distance is less than or equal to 15" to the console to see that it actually triggers.

**When I list Rule1, it says that: "State":"ON"**
`00:34:18.461 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":75,"Free":436,"Rules":"ON VL53L0X#Distance<=15 DO Log \"Distance is less than or equal to 15\" ENDON"}}`

The below is then the order, how I have implemented the Rule1 and waiting for a Distance<=15 but there is no trigger that shows the message string **"Distance is less than or equal to 15"**

I am not sure if I have done this correctly?

**Console Data:**
```
00:28:30.149 RSL: RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":75,"Free":436,"Rules":"ON VL53L0X#Distance<=15 DO Log \"Distance is less than or equal to 15\" ENDON"}}
00:28:30.397 CFG: Saved to flash at F8, Count 60, Bytes 4096
00:28:38.260 RSL: STATE = {"Time":"2023-10-12T00:28:38","Uptime":"0T06:30:02","UptimeSec":23402,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"House_A","BSSId":"D4:CA:6D:AB:8F:64","Channel":10,"Mode":"11n","RSSI":78,"Signal":-61,"LinkCount":1,"Downtime":"0T00:00:03"}}
00:28:38.266 RSL: SENSOR = {"Time":"2023-10-12T00:28:38","VL53L1X":{"Distance":21.3}}
00:28:48.258 RSL: STATE = {"Time":"2023-10-12T00:28:48","Uptime":"0T06:30:12","UptimeSec":23412,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"House_A","BSSId":"D4:CA:6D:AB:8F:64","Channel":10,"Mode":"11n","RSSI":80,"Signal":-60,"LinkCount":1,"Downtime":"0T00:00:03"}}
00:28:48.265 RSL: SENSOR = {"Time":"2023-10-12T00:28:48","VL53L1X":{"Distance":21.5}}
00:28:58.250 RSL: STATE = {"Time":"2023-10-12T00:28:58","Uptime":"0T06:30:22","UptimeSec":23422,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"House_A","BSSId":"D4:CA:6D:AB:8F:64","Channel":10,"Mode":"11n","RSSI":76,"Signal":-62,"LinkCount":1,"Downtime":"0T00:00:03"}}
00:28:58.256 RSL: SENSOR = {"Time":"2023-10-12T00:28:58","VL53L1X":{"Distance":10.5}}
00:29:08.279 RSL: STATE = {"Time":"2023-10-12T00:29:08","Uptime":"0T06:30:32","UptimeSec":23432,"Heap":24,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"House_A","BSSId":"D4:CA:6D:AB:8F:64","Channel":10,"Mode":"11n","RSSI":74,"Signal":-63,"LinkCount":1,"Downtime":"0T00:00:03"}}
00:29:08.286 RSL: SENSOR = {"Time":"2023-10-12T00:29:08","VL53L1X":{"Distance":10.3}}
```

Thank you!

**Comment by sfromis** [comment]
It may help to use the right sensor name in the rule :wink: 

There exists no Tasmota command name "Log", but it should still try, including printing the bad command.

**Reply by jackpotcityco** [reply]
Thank you! how could I miss that, it must be: "VL53L1X" and not "VL53L0X". It now works when I changed to the correct sensor :-)

Yes you are right, I managed to write the command like this instead as 2 conditions:
`Rule1 ON VL53L1X#Distance<=15 DO ConsoleLog,DistanceA:%value% ENDON ON VL53L1X#Distance>=30 DO ConsoleLog,DistanceB:%value% ENDON`

Now the Console is reacting on both "DistanceA" and "DistanceB". Only thing I might wonder is for the: `Command":"Unknown"` if that is normal or if I did something wrong in the Rule1?
`RESULT = {"Command":"Unknown"}`

**Output from the console:**
```
14:08:25.409 RSL: STATE = {"Time":"2023-10-12T14:08:25","Uptime":"0T00:27:30","UptimeSec":1650,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"Wifi":{"AP":1,"SSId":"House_A","BSSId":"D4:CA:6D:AB:8F:64","Channel":10,"Mode":"11n","RSSI":66,"Signal":-67,"LinkCount":1,"Downtime":"0T00:00:09"}}
14:08:25.415 RSL: SENSOR = {"Time":"2023-10-12T14:08:25","VL53L1X":{"Distance":12.9}}
14:08:25.537 RUL: VL53L1X#DISTANCE<=15 performs "ConsoleLog,DistanceA:12.9"
14:08:25.538 SRC: Rule
14:08:25.541 CMD: Grp 0, Cmd 'CONSOLELOG', Idx 1, Len 15, Pld -99, Data ',DistanceA:12.9'
14:08:25.545 RSL: RESULT = {"Command":"Unknown"}
14:08:26.242 RUL: VL53L1X#DISTANCE<=15 performs "ConsoleLog,DistanceA:12.9"
14:08:26.244 SRC: Rule
14:08:26.246 CMD: Grp 0, Cmd 'CONSOLELOG', Idx 1, Len 15, Pld -99, Data ',DistanceA:12.9'
14:08:26.251 RSL: RESULT = {"Command":"Unknown"}
```

**Reply by sfromis** [reply, marked as answer]
It is completely normal that you get `{"Command":"Unknown"}` when you issue an unknown command. Yes, you did "something wrong" in not using a valid command.

One way of doing it with "fewer complaints" would be something like `var1 blah blah`, assigning something to a variable, and then this will be printed as a "result".

**Reply by jackpotcityco** [reply]
Thank you sfromis, then I understand. Nice to get this working!

Thank you for your help!
